blockindex/contractstaking/bucket.go

```diff
@@ -6,16 +6,19 @@
 package contractstaking
 
 import (
+	"time"
+
 	"github.com/iotexproject/iotex-core/action/protocol/staking"
 )
 
 // Bucket defines the bucket struct for contract staking
 type Bucket = staking.VoteBucket
 
-func assembleBucket(token uint64, bi *bucketInfo, bt *BucketType, contractAddr string) *Bucket {
+func assembleBucket(token uint64, bi *bucketInfo, bt *BucketType, contractAddr string, blockInterval time.Duration) *Bucket {
 	vb := Bucket{
 		Index:                     token,
 		StakedAmount:              bt.Amount,
+		StakedDuration:            time.Duration(bt.Duration) * blockInterval,
```

Comment:
pls make sure all NFT buckets are processed by this `assembleBucket` before being passed to `CalculateVoteWeight`